src/sql-editor/components/SQLEditor.tsx

```diff
@@ -296,6 +300,7 @@ function initializeLanguageRegistries(id: string) {
       operators: new Registry(initOperatorsRegistry),
       suggestionKinds: new Registry(initSuggestionsKindRegistry),
       positionResolvers: new Registry(initStatementPositionResolvers),
+      macros: new Registry(initMacrosRegistry),
```

Comment:
Cool, we also need to provide an API for the SQLEditor user to provide data source specific macros.